pkg/eval/builtin_fn_fs_test.go

```diff
@@ -23,3 +32,36 @@ func TestBuiltinFnFS(t *testing.T) {
 		That(`-is-dir ~/file`).Puts(false),
 	)
 }
+
+func TestBuiltinCd(t *testing.T) {
+	tmpHome, cleanup := InTempHome()
+	defer cleanup()
+
+	mustMkdirAll("d1")
+	mustMkdirAll("d2")
+	d1Path := tmpHome + string(os.PathSeparator) + "d1"
+	d2Path := tmpHome + string(os.PathSeparator) + "d2"
+	os.Chdir(d1Path)
+	os.Setenv("HOME", d1Path)
+
+	// We install this mock for all tests, not just the one that needs it,
+	// because it should not be invoked by any of the other tests.
+	util.CurrentUser = currentUser
+	defer func() { util.CurrentUser = user.Current }()
+
+	Test(t,
+		That(`old = $pwd; cd `+d2Path+`; put $pwd; cd $old; eq $old $pwd`).Puts(d2Path, true),
+		That(`old = $pwd; cd /; cd `+d1Path+`; eq $pwd $E:HOME; cd $old`).Puts(true),
```

Comment:
It's not clear which aspects of the behavior of `cd` these two tests intend to test; it would be useful to document that in comments.

Reply:
Good point. Done.